include/lug/System/Memory.inl

```diff
@@ -0,0 +1,154 @@
+template <typename T, class Arena, class ...Args>
+inline T* new_one(size_t alignment, const char* file, size_t line, Arena& arena, Args&&... args) {
+    return new (arena.allocate(sizeof(T), alignment, 0, file, line)) T{std::forward<Args>(args)...};
+}
+
+template <typename T, class Arena>
+inline void delete_one(T* object, Arena& arena) {
+    object->~T();
+    arena.free(object);
+}
+
+template <typename T, class Arena, class ...Args, typename std::enable_if<!std::is_pod<T>::value, int>::type>
+inline T* new_array(size_t alignment, size_t nb, const char* file, size_t line, Arena& arena, Args&&... args) {
+    void* const ptr = arena.allocate(sizeof(T) * nb + sizeof(size_t), alignment, alignof(size_t), file, line);
+
+    if (!ptr) {
+        return nullptr;
+    }
+
+    // Store the size of the array
+    size_t* size_ptr = static_cast<size_t*>(ptr);
+    *size_ptr = nb;
+
+    // Call the constructors
+    T* const user_ptr = reinterpret_cast<T*>(size_ptr + 1);
+    for (size_t i = 0; i < nb; ++i)
```

Comment:
If you decided to add brackets to single line `ifs`, do so for single line `for`